lib/perimeterx.js

```diff
@@ -32,167 +34,13 @@ function initPXModule(params) {
-    const pxutil = require('./utils/pxutil');
-    const pxCaptcha = require('./utils/pxcaptcha');
-
-    if (!pxconfig.ENABLE_MODULE || pxutil.checkForStatic(req, pxconfig.STATIC_FILES_EXT)) {
-        return next();
-    }
-    try {
-        const pxCtx = new PxContext(pxconfig, req);
-        if (req.cookies && req.cookies['_pxCaptcha']) {
-            const pxcptch = req.cookies['_pxCaptcha'];
-            res.clearCookie('_pxCaptcha', {});
-            pxCaptcha.verifyCaptcha(pxCtx, pxcptch, (err, result) => {
-                if (err) {
-                    if (err === "timeout") {
-                        pxLogger.debug('sending page requested activity for captcha timeout');
-                        pxCtx.passReason = pxconfig.PASS_REASON.CAPTCHA_TIMEOUT
-                        return pxPass(pxCtx);
-                    } else {
-                        pxLogger.error('error while evaluation perimeterx captcha');
-                        pxCtx.blockReason = 'captcha_verification_failed';
-                        return pxBlock(pxCtx);
-                    }
-                }
-
-                if (!result || result.status === -1) {
-                    pxLogger.debug('perimeterx captcha verification faild');
-                    pxCtx.blockReason = 'captcha_verification_failed';
-                    return pxBlock(pxCtx);
-                }
-
-                pxLogger.debug('sending page requested activity from captcha');
-                pxCtx.passReason = pxconfig.PASS_REASON.CAPTCHA;
-                pxPass(pxCtx);
-                return next();
-
-            });
-        } else {
-            pxutil.verifyUserScore(pxCtx, (action) => {
-                pxLogger.debug('score action ' + action);
-
-                // check for additional activity handler
-                if (pxconfig.ADDITIONAL_ACTIVITY_HANDLER) {
-                    pxconfig.ADDITIONAL_ACTIVITY_HANDLER(pxCtx, pxconfig);
-                }
-
-                if (pxconfig.CUSTOM_REQUEST_HANDLER){
-                    pxconfig.CUSTOM_REQUEST_HANDLER(pxCtx, pxconfig, req, res);
-                    if (res.headersSent) {
-                        return;
-                    }
-                }
-                if (pxCtx.score < pxconfig.BLOCKING_SCORE || pxconfig.MODULE_MODE === pxconfig.MONITOR_MODE.MONITOR) {
-                    pxPass(pxCtx);
-                    return next();
-                } else {
-                    pxBlock(pxCtx);
-
-                    generateTemplate(pxCtx, pxconfig, function(htmlTemplate){
-                        res.status(403);
-                        res.header('Content-Type', 'text/html');
-                        res.send(htmlTemplate);
-                        res.end();
-                    });
-                }
-            });
+    enforcer.enforce(req, res, (response) => {
+        if (response) { //block
+            res.status(response.status);
+            res.header('Content-Type', 'text/html');
```

Comment:
if this is the last line for rendering the block page
what happens on mobile sdk? application json should be rendered

Reply:
No support (yet) for mobile on node enforcer